### ISSUE TITLE
Small issue with function cisco_comm_indexing for Cisco C3750X devices

### ISSUE THREAD

**Issue by earendilfr** [netdisco/snmp-info#266]
I have a switch Cisco C3750X with is registered in Netdisco like a SNMP::Info::Layer3::C6500.
When I perform a macsuck, it doesn't work as expected (it perform only the native VLAN) because the call to the function cisco_comm_indexing always call the function in class SNMP::Info and not the function in class SNMP::Info::Layer3::CiscoSwitch

## Expected Behavior
Return 1 to function cisco_comm_indexing() (by using the function in class SNMP::Info::Layer3::CiscoSwitch

## Current Behavior
return 0 with usage of the function cisco_comm_indexing from the class SNMP::Info 

## Possible Solution
From my point of view, I have just add a function cisco_comm_indexing which always return 1 in class SNMP::Info::Layer3::C6500 to solve my issue.

## Context
In Netdisco, I have a lot of device that doesn't not correctly perform the macsuck functionnality due to this issue.

## Your Environment
* SNMP::Info version used: 3.56
* Netdisco version (if used): 2.39.24

**Comment by ollyg** [comment]
I wonder if Layer3/Cisco.pm is failing to find `vtpVersion` and therefore returning false, as per

https://github.com/netdisco/snmp-info/blob/a1769aa9050972d199e8835a688d3301d7ed4e10/lib/SNMP/Info/Layer3/Cisco.pm#L141

@earendilfr can you please provide the output of:

`bin/netdisco-do show -d 1.2.3.4 -DI -e vtpVersion`

**Comment by earendilfr** [comment]
You can found the result of the command:
```
$ netdisco-do show -d 1.2.3.4 -DI -e vtpVersion
[7126] 2018-05-28 15:11:15  info App::Netdisco version 2.039024 loaded.
[7126] 2018-05-28 15:11:15  info show: [1.2.3.4]/vtpVersion started at Mon May 28 17:11:15 2018
[7126] 2018-05-28 15:11:15 debug show: running with timeout 600s
[7126] 2018-05-28 15:11:15 debug => running workers for phase: check
[7126] 2018-05-28 15:11:15 debug -> run worker check/_base_/0
[7126] 2018-05-28 15:11:15 debug Show is able to run
[7126] 2018-05-28 15:11:15 debug => running workers for phase: main
[7126] 2018-05-28 15:11:15 debug -> run worker main/_base_/100
[7126] 2018-05-28 15:11:15 debug snmp reader cache warm: [1.2.3.4]
[7126] 2018-05-28 15:11:15 debug [1.2.3.4:161] try_connect with ver: 3, class: SNMP::Info::Layer3::C6500, comm: <hidden>
SNMP::Info::_global uptime : DISMAN-EVENT-MIB::sysUpTimeInstance : .1.3.6.1.2.1.1.3.0
SNMP::Info::_global layers : SNMPv2-MIB::sysServices.0 : .1.3.6.1.2.1.1.7.0
SNMP::Info::_global description : SNMPv2-MIB::sysDescr.0 : .1.3.6.1.2.1.1.1.0
SNMP::Info::_global id : SNMPv2-MIB::sysObjectID.0 : .1.3.6.1.2.1.1.2.0
SNMP::Info 3.56
SNMP::Info::device_type() layers:00000110 id:9 sysDescr:"Cisco IOS Software, C3750E Software (C3750E-UNIVERSALK9-M), Version 15.0(2)SE10a, RELEASE SOFTWARE (fc3) Technical Support: http://www.cisco.com/techsupport Copyright (c) 1986-2016 by Cisco Systems, Inc. Compiled Thu 03-Nov-16 13:36 by prod_rel_team"
SNMP::Info::_global vtpVersion : CISCO-VTP-MIB::vtpVersion.0 : .1.3.6.1.4.1.9.9.46.1.1.1.0
"three"
[7126] 2018-05-28 15:11:16  info show: finished at Mon May 28 17:11:16 2018
[7126] 2018-05-28 15:11:16  info show: status done: Showed vtpVersion response from 1.2.3.4
```

And you can also found the result of the command:
- for cisco_comm_indexing without my change on SNMP/Info/Layer3/C6500.pm file:
```
netdisco-do show -d 1.2.3.4 -e cisco_comm_indexing
[7118] 2018-05-28 15:10:52  info App::Netdisco version 2.039024 loaded.
[7118] 2018-05-28 15:10:52  info show: [1.2.3.4]/cisco_comm_indexing started at Mon May 28 17:10:52 2018
undef
[7118] 2018-05-28 15:10:53  info show: finished at Mon May 28 17:10:53 2018
[7118] 2018-05-28 15:10:53  info show: status done: Showed cisco_comm_indexing response from 1.2.3.4
```

- for cisco_comm_indexing with my change on SNMP/Info/Layer3/C6500.pm file:
```
$ netdisco-do show -d 1.2.3.4 -e cisco_comm_indexing
[7275] 2018-05-28 15:14:48  info App::Netdisco version 2.039024 loaded.
[7275] 2018-05-28 15:14:48  info show: [1.2.3.4]/cisco_comm_indexing started at Mon May 28 17:14:48 2018
1
[7275] 2018-05-28 15:14:49  info show: finished at Mon May 28 17:14:49 2018
[7275] 2018-05-28 15:14:49  info show: status done: Showed cisco_comm_indexing response from 1.2.3.4
```

**Comment by earendilfr** [comment]
Second point, if I read correctly, I have:

my switch is defined as a SNMP::Info::Layer3::C6500
that depend of SNMP::Info::Layer3::CiscoSwitch
And, in this class, I have a function cisco_comm_indexing that should return 1. 

Why did I never use this function?

**Comment by jeneric** [comment]
Please upgrade to the latest version of SNMP:Info, I believe this issue was resolved in SNMP::Info version 3.57 and this commit 7221797826a63b96e613159f12f3a64aef82bf27
